app/models/pet.rb

```diff
@@ -10,4 +12,8 @@ def shelter_name
   def self.adoptable
     where(adoptable: true)
   end
+
+  def self.search(search_params) 
```

Comment:
This is a brownfield project, so you should look around at the existing code and see what code you could leverage for your needs! In this case, there's an already written search method in the ApplicationRecord parent class, which is inherited by every model. That means that you can call any model class with `.search` and get a search output. Your query looks great, but just wanted to point out that this code is already written for you!